tests/privacy_scenario_3_test.py

```diff
@@ -0,0 +1,265 @@
+    Print("OK: Node2 didn't advance. head still at {}".format(apiNode2_head))
+    assert apiNode2_InSync
+
+    # after last recorded block number (rbn4) is made irreversible, verify
+    # node1 and node2 do not advance their LIB to rbn4 and do not continue to receive blocks
+    waitSec = 6
+    Print("Waiting for {} seconds to ensure api nodes LIB doesn't move".format(waitSec))
+    time.sleep(waitSec)
+
+    Print("Node1 LIB = {}".format(apiNode1.getIrreversibleBlockNum()))
+    assert apiNode1.getIrreversibleBlockNum() < trans_block_num
+    Print("Node2 LIB = {}".format(apiNode2.getIrreversibleBlockNum()))
+    assert apiNode2.getIrreversibleBlockNum() < trans_block_num
+
+    # add node1 back to security group *
+    trans = securityGroup.editSecurityGroup(addNodes=[apiNode1])
+    # record block number (rbn5) transaction was added in
+    trans_block_num = trans["processed"]["block_num"]
+
+    # before last recorded block number (rbn5) is made irreversible, verify
+    # node1 and node2 do not receive blocks
+    #comparing here with trans_block_num - 12 to avoid possible flakiness when LIB advances inside while
+    while prod1.getIrreversibleBlockNum() <= trans_block_num - 12:
+        assert apiNode1.getHeadBlockNum() == apiNode1_head
+        assert apiNode2.getHeadBlockNum() == apiNode2_head
+        Print("Node1 head = {}".format(apiNode1.getHeadBlockNum()))
+        Print("Node2 head = {}".format(apiNode2.getHeadBlockNum()))
+        time.sleep(0.5)
+
+    # after last recorded block number (rbn5) is made irreversible, verify
+    # node1 and node2 start to receive blocks
+    assert apiNode1.waitForLibToAdvance()
+    Print("After {} became LIB Node1 head = {} Node2 head = {}".format(trans_block_num, apiNode1.getHeadBlockNum(), apiNode2.getHeadBlockNum()))
+    #######################################################################
+    # Workaround.
+    # Restart shouldn't be needed here but due to bug in net_plugin we have to.
+    # this supposed to be fixed in nodeos and removed
+    apiNode2.kill(signal.SIGTERM)
+    apiNode2.relaunch(cachePopen=True)
+    #######################################################################
+    apiNode2.waitForBlock(trans_block_num, timeout=240, blockType=BlockType.lib, reportInterval=1)
+    Print("apiNode2 LIB block is now {}".format(trans_block_num))
+    apiNode2.waitForHeadToAdvance()
+
+    # remove node2 back to security group *
+    trans = securityGroup.editSecurityGroup(removeNodes=[apiNode2])
+    # record block number (rbn6) transaction was added in
+    trans_block_num = trans["processed"]["block_num"]
+
+    # after node1’s LIB advances to rbn5, verify
+    # node1 and node2 continue to sync
+    apiNode2_head = apiNode2.getHeadBlockNum()
+    Print("Node1 head = {} Node2 head = {}".format(apiNode2_head, apiNode1.getHeadBlockNum()))
+    apiNode2_Advancing = False
+    while apiNode1.getIrreversibleBlockNum() < trans_block_num:
+        temp = apiNode2.getHeadBlockNum()
```

Comment:
change to get data from apiNode2 before you get if from apiNode1, like above comments

Reply:
that shouldn't result in flakiness as we have few seconds until API Node 1 LIB will be ahead of `trans_block_num` but I'll do just in case